modules/platforms/dotnet/Apache.Ignite.Core/Impl/Services/Services.cs

```diff
@@ -445,7 +445,7 @@ public dynamic GetDynamicServiceProxy(string name, bool sticky)
             }
             finally
             {
-                Marshaller.RegisterSameJavaType.Value = false;
+                Marshaller.RegisterSameJavaTypeTl.Value = false;
```

Comment:
Let's preserve the previous thread-local value instead of assuming it to be false. Nested .NET -> Java -> .NET -> Java -> ... calls are possible.

Reply:
Fixed